src/openms/source/ANALYSIS/OPENSWATH/PeakIntegrator.cpp

```diff
@@ -145,21 +168,26 @@ namespace OpenMS
         peak_apex_pos = it->getPos();
       }
     }
+
     if (integration_type_ == INTEGRATION_TYPE_TRAPEZOID)
     {
-      for (auto it = p.PosBegin(left); it != p.PosEnd(right) - 1; ++it)
-      {
-        peak_area += ((it + 1)->getPos() - it->getPos()) * ((it->getIntensity() + (it + 1)->getIntensity()) / 2.0);
-      }
+      peak_area = compute_peak_area_trapezoid(left, right);
     }
     else if (integration_type_ == INTEGRATION_TYPE_SIMPSON)
     {
-      if (n_points < 3)
+      if (n_points == 1)
```

Comment:
Is there a similar check for Trapezoid integration?

Reply:
No, adding it :+1: